test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -686,6 +686,61 @@ public void NodeDescriptionDeserilizationTest()
             var node = this.CurrentDynamoModel.CurrentWorkspace.Nodes.First();
             Assert.AreEqual(node.Description, "Makes a new list out of the given inputs");
         }
+        [Test]
+        public void OutPortDescriptionDeserilizationTest()
+        {
+            //similar to above test, uses a modified graph to assert port tooltips not deserialized when possible.
+            var testFile = Path.Combine(TestDirectory, @"core\serialization\PortTooltipDeserilizationTest.dyn");
+            OpenModel(testFile);
+
+            var ztNode = this.CurrentDynamoModel.CurrentWorkspace.Nodes.Where(x => x.GUID == new Guid("bda3e3e4c18c461dae5598df465035b2")).First();
+            Assert.AreEqual(ztNode.OutPorts.First().ToolTip, "An Arc");
+
+            var nodeModelNode = this.CurrentDynamoModel.CurrentWorkspace.Nodes.Where(x => x.GUID == new Guid("c848cc3cb24a477f8248e53fc9304cc1")).First();
+            Assert.AreEqual(nodeModelNode.OutPorts.First().ToolTip, "Selected colors");
+        }
+
+        [Test]
+        [Category("Failure")]
+        [Category("TechDebt")]
+        public void OutPortDescriptionDeserilizationTest_VariableInputNodes()
+        {
+            //similar to above test, uses a modified graph to assert port tooltips not deserialized when possible.
+            var testFile = Path.Combine(TestDirectory, @"core\serialization\PortTooltipDeserilizationTest.dyn");
+            OpenModel(testFile);
+            var variableInputNode = this.CurrentDynamoModel.CurrentWorkspace.Nodes.Where(x => x.GUID == new Guid("db8fd7b97be1413e91897316ae75b51a")).First();
+            Assert.AreEqual(variableInputNode.OutPorts.First().ToolTip, "Combined lists");
```

Comment:
Is this for a future task to handle tooltips for dynamically added ports?

Reply:
yep exactly.